File: 动态规划/1.爬楼梯/1.dfs搜索/python/main.py
from typing import List


class Solution:
    def climbStairs(self, n: int) -> int:
        if n <= 2:
            return n
        ans = 0

        def dfs(index):
            nonlocal ans
            nonlocal n
            if index >= n:
                if index == n:
                    ans += 1
                return
            for i in range(1, 3):
                dfs(index + i)

        dfs(0)
        return ans


if __name__ == '__main__':
    s = Solution()

File: 动态规划/1.爬楼梯/1.dfs搜索/java/Main.java
class Solution {
    int ans = 0;
    public void dfs(int index,int n){
        if(index>n){
            return;
        }
        if(index==n){
            ans++;
            return;
        }
        for(int i=1;i<=2;i++){
            dfs(index+i,n);
        }
    }
    public int climbStairs(int n) {
        if(n<=2){
            return n;
        }
        dfs(0,n);    
        return this.ans;
    }
}